staging/src/k8s.io/cloud-provider/controllers/service/controller.go

```diff
@@ -970,6 +975,10 @@ func nodeIncludedPredicate(node *v1.Node) bool {
 	return !hasExcludeBalancerLabel
 }
 
+func nodeHasProviderIDPredicate(node *v1.Node) bool {
+	return node.Spec.ProviderID != ""
```

Comment:
Is there no case where someone runs without this set?  It is strictly optional.

Reply:
Hmm, good question...I assumed that to be the case, but I can't tell for sure :/ 

The safest might be to just trigger a resync in `shouldSyncUpdatedNode` due to changes to the field, but not actually require that nodes have the field set?